src/lib/InstantSearch.js

```diff
@@ -130,7 +142,9 @@ class InstantSearch extends EventEmitter {
     widgets.forEach(widget => {
       // Add the widget to the list of widget
       if (widget.render === undefined && widget.init === undefined) {
-        throw new Error('Widget definition missing render or init method');
+        throw new Error(`The widget definition expects a \`render\` and/or an \`init\` method.
+
+See: http://alg.li/isjs-custom-widgets`);
```

Comment:
I still think this is going to be hard to manage if the page in the doc gets moved or refactored.
Short is good for bandwidth but it's even harder to grep through pages we moved in the docs now that they are shortened.

Reply:
At some point we need to make a decision. We've discussed many approaches and didn't come to a conclusion yet.

(1) If we start adopting the shorten-URL pattern, we need to educate the team to connect to short.cm and create a shorter URL. We don't need to update the source code if the link changes but only in the short.cm dashboard. Another negligible plus is that the links are shorter and therefore the bundle size get's slightly lighter.

(2) If we inline all the links, we need to update the source code whenever a link breaks (this was the initial solution of this PR).

(3) If we don't include links, errors are completely meaningless for the users.

I don't mind going for either (1) or (2).

Reply:
agreed with you Francois, and the links are easiest. Maybe later some way to make it more automatic (e.g. generated from some GitHub file or something, but that wouldn't change too much IMO)